objects/src/accounts/component/template/storage_entry/mod.rs

```diff
@@ -0,0 +1,715 @@
+                .map(|word_repr| {
+                    word_repr.clone().try_build_word(init_storage_data).map(StorageSlot::Value)
+                })
+                .collect::<Result<Vec<StorageSlot>, _>>()?),
+        }
+    }
+}
+
+// SERIALIZATION
+// ================================================================================================
+
+impl Serializable for StorageEntry {
+    fn write_into<W: ByteWriter>(&self, target: &mut W) {
+        match self {
+            StorageEntry::Value { name, description, slot, value } => {
+                target.write_u8(0u8);
+                target.write(name);
+                target.write(description);
+                target.write_u8(*slot);
+                target.write(value);
+            },
+            StorageEntry::Map { name, description, slot, values } => {
+                target.write_u8(1u8);
+                target.write(name);
+                target.write(description);
+                target.write_u8(*slot);
+                target.write(values);
+            },
+            StorageEntry::MultiSlot { name, description, slots, values } => {
+                target.write_u8(2u8);
+                target.write(name);
+                target.write(description);
+                target.write(slots);
+                target.write(values);
+            },
+        }
+    }
+}
+
+impl Deserializable for StorageEntry {
+    fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
+        let variant_tag = source.read_u8()?;
+        let name: String = source.read()?;
+        let description: Option<String> = source.read()?;
+
+        match variant_tag {
+            // Value
+            0 => {
+                let slot = source.read_u8()?;
+                let value: WordRepresentation = source.read()?;
+
+                Ok(StorageEntry::Value { name, description, slot, value })
+            },
+
+            // Map
+            1 => {
+                let slot = source.read_u8()?;
+                let values: Vec<MapEntry> = source.read()?;
+
+                Ok(StorageEntry::Map { name, description, slot, values })
+            },
+
+            // MultiSlot
+            2 => {
+                let slots: Vec<u8> = source.read()?;
+                let values: Vec<WordRepresentation> = source.read()?;
+
+                Ok(StorageEntry::MultiSlot { name, description, slots, values })
+            },
+
+            // Unknown tag => error
+            _ => Err(DeserializationError::InvalidValue(format!(
+                "unknown variant tag for StorageEntry: {}",
+                variant_tag
+            ))),
+        }
+    }
+}
+
+// SERDE SERIALIZATION
+// ================================================================================================
+
+/// Used as a helper for validating and (de)serializing storage entries
+#[derive(Default)]
+#[cfg_attr(feature = "std", derive(serde::Deserialize, serde::Serialize))]
+struct RawStorageEntry {
```

Comment:
This is used specifically for TOML serialization, right? I would potentially move this and related structs (e.g., `StorageValues`) into a separate module. Or maybe we should have a module that deals with TOML serialization for various structs. Could be something like `src/accounts/component/template/storage/toml.rs`.

Reply:
Yes, was thinking of this as well originally. I moved everything to a `toml` submodule